src/main/kotlin/org/rust/cargo/toolchain/RsToolchainBase.kt

```diff
@@ -126,11 +129,24 @@ abstract class RsToolchainBase(val location: Path) {
         const val RUSTC_BOOTSTRAP: String = "RUSTC_BOOTSTRAP"
         const val RUSTC_WRAPPER: String = "RUSTC_WRAPPER"
 
-        fun suggest(): RsToolchainBase? =
-            RsToolchainFlavor.getApplicableFlavors()
+        @JvmOverloads
+        fun suggest(projectDir: Path? = null): RsToolchainBase? {
+            val path = projectDir?.toString() ?: return null
+            if (!WslDistributionManager.isWslPath(path)) return null
+
+            val distribution = WslPath.getDistributionByWindowsUncPath(path)
+            val toolchain = distribution
+                ?.getHomePathCandidates()
+                ?.filter { RsToolchainFlavor.getFlavor(it) != null }
+                ?.mapNotNull { RsToolchainProvider.getToolchain(it.toAbsolutePath()) }
+                ?.firstOrNull()
+            if (toolchain != null) return toolchain
```

Comment:
Now, if you pass `null` as `projectDir` or pass non-wsl path, we always return null as suggested toolchain

Reply:
Fixed.